src/crons/XKCD.ts

```diff
@@ -0,0 +1,82 @@
+import { Cron, findTextChannelByName } from '../framework';
+import got from 'got';
+import { parse } from 'node-html-parser';
+import { decode } from 'html-entities';
+import { KeyValue } from '../database';
+import { EmbedBuilder } from 'discord.js';
+
+const DATETIME_FMT: Intl.DateTimeFormatOptions = {
+  timeZone: 'Europe/Paris',
+  year: 'numeric',
+  month: 'long',
+  day: 'numeric',
+  hour: 'numeric',
+  minute: 'numeric',
+};
+
+export default new Cron({
+  enabled: true,
+  name: 'XKCD',
+  description:
+    'Vérifie toutes les 30 minutes si un nouveau strip de XKCD est sortis',
+  // schedule: '5,35 * * * *',
+  schedule: '* * * * *',
+  async handle(context) {
+    const strip = await getLastXKCDStrip();
+
+    // vérifie le strip trouvé avec la dernière entrée
+    const lastStrip = await KeyValue.get<string>('Last-Cron-XKCD');
+    const stripStoreIdentity = strip?.id ?? null;
+    if (lastStrip === stripStoreIdentity) return; // skip si identique
+
+    await KeyValue.set('Last-Cron-XKCD', stripStoreIdentity); // met à jour sinon
+
+    if (!strip) return; // skip si pas de strip
+
+    context.logger.info(`Found a new XKCD strip`, strip);
+
+    const channel = findTextChannelByName(context.client.channels, 'gif');
+
+    await channel.send({
+      embeds: [
+        new EmbedBuilder()
+          .setURL(strip.link)
+          .setTitle(strip.title)
+          .setImage(strip.imageUrl)
+          .setTimestamp(strip.date)
+          .setFooter({ text: strip.description }),
+      ],
+    });
+  },
+});
+
+interface IXKCDStrip {
+  id: string;
+  link: string;
+  title: string;
+  description: string;
+  date: Date;
+  imageUrl: string;
+}
+
+export async function getLastXKCDStrip(): Promise<IXKCDStrip | null> {
+  const { body } = await got('https://xkcd.com/atom.xml');
```

Comment:
J'ai pas regardé à quoi ressemble le body, je te fais confiance pour cette fonction.

Reply:
On devrait passer tout payload externe à travers une validation de données, via [`zod`](https://github.com/colinhacks/zod) par exemple.

Reply:
le body est un string contenant le feed en xml au format atom

Reply:
Why not @RomainLanz.